include/maths/CKdTree.h

```diff
@@ -36,6 +36,18 @@ namespace kdtree_detail {
 //! \brief Stubs out the node data parameter for k-d tree.
 struct SEmptyNodeData {};
 
+template<typename T>
+struct unwrapped_ref {
+    using type = T;
+};
+template<typename T>
+struct unwrapped_ref<std::reference_wrapper<T>> {
+    using type = T;
+};
+template<typename T>
+struct unwrapped_ref<std::reference_wrapper<const T>> {
+    using type = T;
+};
```

Comment:
We have some functions like this in `core/UnwrapRef.h`.  Could those be used instead?

Reply:
Good point! I forgot about those.

Reply:
See [dbb74db](https://github.com/elastic/ml-cpp/pull/1070/commits/dbb74dbcd91239394e2ae1bada60489a4cd00cef).